### PR TITLE
Try out upgrading from pypi to git in CI.

### PR DESCRIPTION
**Pull request by mikalstill**

### CODE REVIEW

**Comment by mikalstill** [comment]
I've just realised that renaming interfaces definitely breaks DHCP and probably breaks iptables. Further thought required.

File: Jenkinsfile [hunk 1]
```diff
@@ -36,20 +42,96 @@ pipeline {
...
+                ./deployandtest.sh
+
+                # Log what versions we are running
+                echo "============================================================"
+                pip list | grep shakenfist
+                echo "============================================================"
+          '''
+        }
+      }
+    stage('Create sample infrastructure') {
```
Comment (mandoonandy) [inline Jenkinsfile:62]: Maybe start two of everything? Push a little harder on the upgrade script.

File: deploy/shakenfist_ci/tests/test_upgrades.py [hunk 2]
```diff
@@ -4,170 +4,18 @@
...
-            '192.16.10.0/24', networks['30acadd4-66ee-4390-9477-e31d11fe44cb']['netblock'])
-
-        # Second network
-        self.assertEqual(
-            'nsnet', networks['afce2cce-dec2-4cf7-a61b-2c4614b320c7']['name'])
-        self.assertEqual(
-            'nstest', networks['afce2cce-dec2-4cf7-a61b-2c4614b320c7']['namespace'])
-        self.assertEqual(
-            '192.168.20.0/24', networks['afce2cce-dec2-4cf7-a61b-2c4614b320c7']['netblock'])
+        self.assertIn('upgrade/upgrade', networks)
```
Comment (mandoonandy) [inline deploy/shakenfist_ci/tests/test_upgrades.py:14]: Would be good to test the state of the network.

File: deploy/shakenfist_ci/tests/test_upgrades.py [hunk 3]
```diff
@@ -4,170 +4,18 @@
...
-                'uuid': 'a13ce821-adcb-45a4-9260-1b18b5250ad3',
-                'vdi_port': None,
-                'version': 2,
-                'video': {
-                    'memory': 16384,
-                    'model': 'cirrus'
-                }
-            }, self.system_client.get_instance('a13ce821-adcb-45a4-9260-1b18b5250ad3')
-        )
+            instances['%s/%s' % (inst['namespace'], inst['name'])] = inst
```
Comment (mandoonandy) [inline deploy/shakenfist_ci/tests/test_upgrades.py:19]: Same again. Check the instance has no errors.

**Review by mandoonandy — COMMENTED** [review]
Looks good. I've just suggested a couple of additions on top of the already good test sequence.

File: Jenkinsfile (hunk 1 above)
Comment (mikalstill) [reply Jenkinsfile:62]: Done.

File: deploy/shakenfist_ci/tests/test_upgrades.py (hunk 2 above)
Comment (mikalstill) [reply deploy/shakenfist_ci/tests/test_upgrades.py]: Sorry, done but I didn't reply.

File: deploy/shakenfist_ci/tests/test_upgrades.py (hunk 3 above)
Comment (mikalstill) [reply deploy/shakenfist_ci/tests/test_upgrades.py:22]: Sorry, done but I didn't reply.

**Review by mandoonandy — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 4]
```diff
@@ -6,7 +6,7 @@ pipeline {
   }
 
   stages {
-    stage('Setup') {
+    stage('Dependancies') {
```
[the remark this answers is not in the document]
Comment (mikalstill) [reply Jenkinsfile]: Fixed.

**Review by mandoonandy — APPROVED** [review]
LGTM

**Pull request merged**